src/shoobx/immutable/tests/test_pjpersist.py

```diff
@@ -427,8 +437,17 @@ def test_addRevision(self):
         with Question.__im_create__() as factory:
             q1 = factory('What is the answer')
         self.questions.add(q1)
+
         q2 = q1.__im_clone__()
-        self.questions.addRevision(q2, old=q1)
+        with self.assertRaises(AssertionError):
+            # can NOT add a transient object
+            self.questions.addRevision(q2, old=q1)
+
+        with q1.__im_update__() as q3:
+            q3.answer = 42
```

Comment:
Please make the error case its own test method called `test_addRevision_withTransientObject`.